### PR TITLE
Testing implementation

### PR DESCRIPTION
**Pull request by wezlar**
This branch includes all the test additions and any project adjustments required.

### CODE REVIEW

File: client/pairs.js [hunk 1]
```diff
@@ -6,7 +6,7 @@ import css from './styles/style.styl';
 
 // import components
 import App from './components/App';
-import Game from './components/Game';
+import Game from './components/Game/Game';
```
Comment (dakebl) [inline client/pairs.js]: Should consider renaming `./components/Game/Game.jsx` or `./components/Game/index.jsx`
Then you can still use `import Game from './components/Game';`, as the module resolution will inherit from it's folder name

File: client/utils/newGame/newGame.test.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import React from 'react'
+import { shallow } from 'enzyme'
```
Comment (dakebl) [inline client/utils/newGame/newGame.test.js:2]: Should consider swapping this out for `react-testing-library`: https://github.com/kentcdodds/react-testing-library

Check out this tweet: https://twitter.com/housecor/status/1115265843917725696

File: client/actions/actionCreators.test.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+import mockDeck from '../../testHelpers/__mocks__/mockDeck';
+
+describe(`actionCreators.js`, () => {
+  describe(`actions`, () => {
+    test(`cardFlip()`, () => {
+      const payload = 1;
+      expect(cardFlip(payload)).toMatchSnapshot();
+    });
+    test(`cardMatched()`, () => {
```
Comment (dakebl) [inline client/actions/actionCreators.test.js:20]: Should pad you `test()`'s with new lines.

File: client/components/Card/Card.js [hunk 4]
```diff
@@ -59,44 +59,52 @@ const SVGWrapper = styled.div`
...
+  } = props;
 
-    const file = require(`!!raw-loader!../${image}`);
+  const file = image 
+    ? require(`!!raw-loader!../../${image}`)
+    : '';
 
-    const flippedStyle = isFlipped 
-      ? { transform: 'rotateY(-180deg)' } 
+  const flippedStyle = isFlipped
```
Comment (dakebl) [inline client/components/Card/Card.js]: This is probably short enough to have on one line

File: client/components/Card/Card.test.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
+import React from 'react'
+import { shallow } from 'enzyme'
+import { Card as Component } from './Card';
```
Comment (dakebl) [inline client/components/Card/Card.test.js]: Why did you rename is to `Component`?

**Review by dakebl — CHANGES_REQUESTED** [review]

File: client/components/Card/Card.test.js (hunk 5 above)
Comment (wezlar) [reply client/components/Card/Card.test.js]: Because that is what we do on some tests in Leap2. I'm assuming it so that it's easier to replicate tests for other components without having to rename all the tests.

File: client/pairs.js (hunk 1 above)
Comment (wezlar) [reply client/pairs.js]: I have made this change, but now come across a minor issue in that searching for component files in VS code now becomes slightly more difficult because everything is named `index`

File: client/utils/newGame/newGame.test.js (hunk 2 above)
Comment (wezlar) [reply client/utils/newGame/newGame.test.js:2]: I have added it as a new issue so that I can merge this PR

File: client/components/Card/Card.js (hunk 4 above)
Comment (wezlar) [reply client/components/Card/Card.js]: I have done as suggested. But noticed the only reason I had it like this was to match the others and keep the file looking tidy and readable.
